webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/ViewTabContent.elm

```diff
@@ -452,16 +452,20 @@ tabContent model details =
               groupTreeElem item =
                 let
                   checkIncludeOrExclude : List RuleTarget -> Bool
-                  checkIncludeOrExclude lst = lst
-                    |> List.map (\t -> case t of
-                       NodeGroupId groupId -> groupId
-                       Composition _ _ -> "compo"
-                       Special spe -> spe
-                       Node node -> node
-                       Or _ -> "or"
-                       And _ -> "and"
-                    )
-                    |> List.member item.id
+                  checkIncludeOrExclude lst =
+                    let
+                      getTargetIds : List RuleTarget -> List String
+                      getTargetIds listTargets =
+                        List.concatMap (\target -> case target of
+                           NodeGroupId groupId -> [groupId]
+                           Composition i e -> getTargetIds [i, e]
```

Comment:
Should it be `Composition i e -> getTargetIds [ i ]` when we call `checkIncludeOrExclude includedTargets`,
and `Composition i e -> getTargetIds [ e ]` when we call `checkExcludeOrExclude excludedTargets`, or is it correct like this ?

Reply:
no it's targets form i minus targets from e

a composition is a list of included targets minus items from excluded targets

you should do something 

 let included = getTargets [ i ] 
 excluded = getTargets [e] 
in included |>
      List.filter (i -> List.member i excluded) 

so that member from excluded  are removed from included